image_test.go

```diff
@@ -286,7 +316,30 @@ func TestImageRW(t *testing.T) {
 					}
 
 					if !reflect.DeepEqual(data, table.image) {
-						t.Fatalf("expected image:\nref=%v\ngot=%v", table.image, data)
+						/* uints and ints mixed fails deepequal with equal data
```

Comment:
yeah... well, I wouldn't do it like that.
but my fix is a bit involved:
```
data := reflect.New(reflect.TypeOf(table.image)).Interface()
err = img.Read(data)
```
(and remove the `switch` on bitpix above).

it's okay to do that in a later PR, though.
(in that case, please remove that new test on `[]uint16` and the associated spaghetti code. apologies for the back and forth on this)